TMU/TMU/TMU/TMU/Services/TMU.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * TMU.h
+ *
+ * Created: 2/24/2020 11:51:46 AM
+ *  Author: Ehab & Reem
+ */ 
+
+
+#ifndef TMU_H_
+#define TMU_H_
+
+
+/************************************************************************/
+/*											INCLUDES							            */
+/************************************************************************/
+#include "../Infrastructure/std_types.h"
+#include "../MCAL/Timer.h"
+#include "../MCAL/interrupt.h"
+#include "../Infrastructure/Errors.h"
+
+
+/************************************************************************/
+/*											DEFINES							            */
+/************************************************************************/
+#define BUFFER_SIZE		50
+#define STARTING_INDEX	0
+
+#define INITIAL_VALUE	0
+
+#define PERIODIC			1
+#define ONE_SHOT			2
+
+/* configurations */
+#define TIMER_OVF_VECT			TIMER0_OVF_vect
+#define TIMER_CHANNEL			TIMER_CH0
+#define TIMER_INITIAL_VALUE	0x05
+#define TCNT						TCNT0
+
+#define ERROR_TMU_MODULE_ID	200
+
+
+/************************************************************************/
+/*								  Structures Definitions	                     */
+/************************************************************************/
+typedef struct TMU_Task
+{
+	uint8_t Task_ID;
+	uint8_t periodic_or_oneShot;
+	uint32_t delay_ms;
+	uint32_t current_ticks;
+	void (*CBK_TOV)(void);
```

Comment:
what if i need to extend the feature ? to have user error return as a parameter ?